packages/block-library/src/file/index.js

```diff
@@ -1,241 +1,30 @@
-/**
- * External dependencies
- */
-import { includes } from 'lodash';
-
 /**
  * WordPress dependencies
  */
-import { __, _x } from '@wordpress/i18n';
-import { createBlobURL } from '@wordpress/blob';
-import { createBlock } from '@wordpress/blocks';
-import { select } from '@wordpress/data';
-import { RichText } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import edit from './edit';
 import icon from './icon';
+import metadata from './block.json';
+import save from './save';
+import transforms from './transforms';
 
-export const name = 'core/file';
+const { name } = metadata;
+
+export { metadata, name };
 
 export const settings = {
 	title: __( 'File' ),
-
 	description: __( 'Add a link to a downloadable file.' ),
-
 	icon,
-
-	category: 'common',
-
 	keywords: [ __( 'document' ), __( 'pdf' ) ],
-
-	attributes: {
-		id: {
-			type: 'number',
-		},
-		href: {
-			type: 'string',
-		},
-		fileName: {
-			type: 'string',
-			source: 'html',
-			selector: 'a:not([download])',
-		},
-		// Differs to the href when the block is configured to link to the attachment page
-		textLinkHref: {
-			type: 'string',
-			source: 'attribute',
-			selector: 'a:not([download])',
-			attribute: 'href',
-		},
-		// e.g. `_blank` when the block is configured to open in a new tab
-		textLinkTarget: {
-			type: 'string',
-			source: 'attribute',
-			selector: 'a:not([download])',
-			attribute: 'target',
-		},
-		showDownloadButton: {
-			type: 'boolean',
-			default: true,
-		},
-		downloadButtonText: {
-			type: 'string',
-			source: 'html',
-			selector: 'a[download]',
-			default: _x( 'Download', 'button label' ),
```

Comment:
It was discussed on WordPress.org Slack last week (link requires registration):

https://wordpress.slack.com/archives/C5UNMSU4R/p1554217879115800

The conclusion was that:

> This doesn't really sound like a "default" but an attribute set when the block is first inserted.

To make it work with `block.json` I converted it to the static text. In the previous form, it was confusing enough. It wasn't triggering block validation when editing the same post with a different locale selected, however, it's worth pointing out that the locale of the website may differ from the one that the person editing uses. It's better to make it static and let the author update this value.

Reply:
I guess the decision here was to set this initially to the value of the author's language. In other words, we could use `componentDidMount` or `useEffect` in the `edit` function of the block to set its value when the block is first created. This has the downside of creating an "undo" level maybe but maybe that's fine?

Reply:
I'll give it a go and see how it works 👍

Reply:
> This has the downside of creating an "undo" level maybe but maybe that's fine?

It is precisely what happens. Is there a way to skip adding undo level when calling `setAttributes`? I noticed that this is an issue also when using drag & drop with files. Triggering `undo` puts the block in a bizarre pulsing state in UI.

Reply:
![file-block-undo](https://user-images.githubusercontent.com/699132/55796001-455a8f80-5ac9-11e9-805c-0ff8c1133705.gif)

Reply:
@aduth and @ellatrix any thoughts on this undo behavior for the transient state of the block attributes? Do you know if there is any issue or PR opened which tries to solve it?

Reply:
> @aduth and @ellatrix any thoughts on this undo behavior for the transient state of the block attributes? Do you know if there is any issue or PR opened which tries to solve it?

#8119 is the main issue tracking the problems in this area.

Reply:
Nice, thanks for linking with the issue. It looks like it isn’t a blocker for this PR 👍